x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlCCSUtils.java

```diff
@@ -239,8 +248,6 @@ static void updateExecutionInfoWithClustersWithNoMatchingIndices(
                 }
             } else {
                 if (indexResolution.isValid()) {
-                    // no matching indices and no concrete index requested - just mark it as done, no error
-                    // We check for the valid resolution because if we have empty resolution it's still an error.
                     markClusterWithFinalStateAndNoShards(executionInfo, c, Cluster.Status.SUCCESSFUL, null);
```

Comment:
I think we need _some_ comments here - maybe the above is not true anymore, but then we'd need to replace it with something that is true?